Patching httpfs extension leads to Windows Extensions CI failure

### What happens?

When pushing to DuckDB with a patch to the httpfs extension, this triggers the Windows Extensions CI to fail.

Error message:

```
CMake Error at C:/Program Files/CMake/share/cmake-3.31/Modules/FetchContent.cmake:1918 (message):
Error:   Build step for httpfs_extension_fc failed: 1
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.31/Modules/FetchContent.cmake:1609 (__FetchContent_populateSubbuild)
  C:/Program Files/CMake/share/cmake-3.31/Modules/FetchContent.cmake:2145:EVAL:2 (__FetchContent_doPopulation)
  C:/Program Files/CMake/share/cmake-3.31/Modules/FetchContent.cmake:2145 (cmake_language)
  C:/Program Files/CMake/share/cmake-3.31/Modules/FetchContent.cmake:1978:EVAL:1 (__FetchContent_Populate)
  C:/Program Files/CMake/share/cmake-3.31/Modules/FetchContent.cmake:1978 (cmake_language)
  CMakeLists.txt:1074 (FETCHCONTENT_POPULATE)
  CMakeLists.txt:1183 (register_external_extension)
  .github/config/out_of_tree_extensions.cmake:19 (duckdb_extension_load)
  CMakeLists.txt:1273 (include)


make: *** [Makefile:361: build/extension_configuration/vcpkg.json] Error 1
-- Configuring incomplete, errors occurred!
Error: Process completed with exit code 2.
```

### To Reproduce

You can reproduce this issue by applying a very simple patch to the httpfs extension and then pushing this to DuckDB.

Example code (thanks to @carlopi)

```
cd path/to/duckdb-httpfs
git diff origin/main > path/to/duckdb/.github/patches/extensions/httpfs/sample.patch
cd path/to/duckdb
```

Then change `.github/config/out_of_tree_extensions.cmake` to 

```
duckdb_extension_load(httpfs
    LOAD_TESTS
    GIT_URL https://github.com/duckdb/duckdb-httpfs
    GIT_TAG 85ac4667bcb0d868199e156f8dd918b0278db7b9
    INCLUDE_DIR extension/httpfs/include
    APPLY_PATCHES
    )
```

and push this 

### OS:

iOS 13.3 

### DuckDB Version:

1.2

### DuckDB Client:

CLI

### Hardware:

_No response_

### Full Name:

Lotte Felius

### Affiliation:

CWI

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a source build

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [x] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [x] Yes, I have

--- a/.github/config/out_of_tree_extensions.cmake
+++ b/.github/config/out_of_tree_extensions.cmake
@@ -16,16 +16,12 @@
 #  VCPKG_TARGET_TRIPLET=arm64-osx
 
 ################# HTTPFS
-# Warning: the patching mechanism on windows doesn't work for httpfs somehow.
-# To patch httpfs:
-#  - add patch file, enable APPLY_PATCHES
-#  - disable windows build of httpfs by wrapping in `if (NOT WIN32)`
-#  - IMPORTANT: add a comment that tells people to restore the windows build when removing the patches
 duckdb_extension_load(httpfs
     LOAD_TESTS
     GIT_URL https://github.com/duckdb/duckdb-httpfs
     GIT_TAG 85ac4667bcb0d868199e156f8dd918b0278db7b9
     INCLUDE_DIR extension/httpfs/include
+    APPLY_PATCHES
     )
 
 ### Skip due to missing patch

--- a/.github/patches/extensions/httpfs/httpfs.patch
+++ b/.github/patches/extensions/httpfs/httpfs.patch
@@ -1,0 +1,12 @@
+diff --git a/Makefile b/Makefile
+index ed12193..b556e8f 100644
+--- a/Makefile
++++ b/Makefile
+@@ -1,5 +1,7 @@
+ PROJ_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+ 
++# Hey, I am a patch
++
+ # Configuration of extension
+ EXT_NAME=httpfs
+ EXT_CONFIG=${PROJ_DIR}extension_config.cmake

--- a/scripts/apply_extension_patches.py
+++ b/scripts/apply_extension_patches.py
@@ -35,7 +35,6 @@ if not patches:
     raise_error(error_message)
 
 print(f"Resetting patches in {directory}\n")
-subprocess.run(["git", "log"], check=True)
 subprocess.run(["git", "clean", "-f"], check=True)
 subprocess.run(["git", "reset", "--hard", "HEAD"], check=True)
 # Apply each patch file using patch

